tidy3d/plugins/dispersion/fit_fast.py

```diff
@@ -0,0 +1,836 @@
+"""Fit PoleResidue Dispersion models to optical NK data"""
+
+from __future__ import annotations
+from typing import Tuple, Optional
+
+import numpy as np
+from rich.progress import Progress
+from pydantic import Field, validator, PositiveInt, NonNegativeFloat, PositiveFloat
+import scipy
+
+from .fit import DispersionFitter
+from ...log import log
+from ...components.base import Tidy3dBaseModel, cached_property
+from ...components.medium import PoleResidue
+from ...components.types import ArrayFloat1D, ArrayComplex1D, ArrayFloat2D, ArrayComplex2D
+from ...constants import HBAR
+from ...exceptions import ValidationError
+
+# numerical tolerance for pole relocation for fast fitter
+TOL = 1e-8
+# numerical cutoff for passivity testing
+CUTOFF = np.finfo(np.float32).eps
+# range for checking and enforcing passivity, in addition to extrema method
+PASSIVITY_MIN = -10
+PASSIVITY_MAX = 4
+PASSIVITY_NUM = 1000
+# parameters for passivity optimization
+PASSIVITY_NUM_ITERS_DEFAULT = 50
+SLSQP_CONSTRAINT_SCALE_DEFAULT = 1e35
+# min value of the rms for default weights calculated based on Re[eps], Im[eps]
+RMS_MIN = 0.1
+
+DEFAULT_MAX_POLES = 5
+DEFAULT_NUM_ITERS = 20
+DEFAULT_TOLERANCE_RMS = 1e-5
+
+# this avoids divide by zero errors with lossless poles
+SCALE_FACTOR = 1.01
+
+
+class AdvancedFastFitterParam(Tidy3dBaseModel):
+    """Advanced fast fitter parameters."""
+
+    loss_bounds: Tuple[float, float] = Field(
+        (0, np.inf),
+        title="Loss bounds",
+        description="Bounds (lower, upper) on Im[eps]. Default corresponds to only passivity. "
+        "A lower bound of 0 or greater ensures passivity. To fit a gain medium without "
+        "additional constraints, use ``loss_bounds=(-np.inf, np.inf)``. "
+        "Increasing the lower bound could help with simulation stability. "
+        "A finite upper bound may be helpful when fitting lossless materials. "
+        "In this case, consider also increasing the weight for fitting the imaginary part.",
+    )
+    weights: Tuple[NonNegativeFloat, NonNegativeFloat] = Field(
+        None,
+        title="Weights",
+        description="Weights (real, imag) in objective function for fitting. The weights "
+        "are applied to the real and imaginary parts of the permittivity epsilon. The weights "
+        "will be rescaled together so they average to 1. If ``None``, the weights are calculated "
+        "according to the typical value of the real and imaginary part, so that they are both "
+        "fit well. Changing this can be helpful if fitting either the real or imaginary part is "
```

Comment:
I think it would be better to explain a bit more on how the weights are computed, or what it mean by "they are both fit well".